### PR TITLE
Use openssl 3.0, update aws sdk

### PR DESCRIPTION
**Pull request by dwelch-spike**

### CODE REVIEW

**Comment by dwelch-spike** [comment]
Canceled the test runs for the readme changes. These changes are passing tests.

File: src/s3_api.cc [hunk 1]
```diff
@@ -263,9 +263,8 @@ S3API::_init_api(S3API& s3_api)
 	Aws::InitAPI(s3_api.options);
 
 	Aws::Client::ClientConfiguration conf;
-	if (s3_api.region.empty() && s3_api.endpoint.empty()) {
-		err("S3 API must be enabled by specifying a region if no endpoint "
-				"override is given.");
+	if (s3_api.region.empty()) {
+		err("S3 API must be enabled by specifying a region.");
```
Comment (jdogmcsteezy) [inline src/s3_api.cc]: I guess a region is always required now and before providing a URL was enough? Is that considered breaking? Maybe there is a way around this

Comment (jdogmcsteezy) [reply src/s3_api.cc]: maybe it was a bug?

**Review by jdogmcsteezy — APPROVED** [review]
LGTM, but a question that might affect the next release version

File: src/s3_api.cc (hunk 1 above)
Comment (dwelch-spike) [reply src/s3_api.cc]: I couldn't find any mention of this when I upgraded but I see something that reverts it in aws-cpp-sdk 1.10.55 https://github.com/aws/aws-sdk-cpp/pull/2295 . I upgraded to that and confirmed that no region works again.

**Pull request merged**